src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvOutputConsumer.cs

```diff
@@ -82,6 +88,17 @@ public class LibuvOutputConsumer
             }
         }
 
+        private static void DisposeSocket(Exception ex, object state)
+        {
+            // If the pipe writer was closed with a TimeoutException, a write timed out.
+            // Disposing the socket will cancel the pending UvWriteReq.
+            if (ex is TimeoutException)
```

Comment:
I think it's reasonable to have the contract be to close the socket immediate if the output/response pipe is completed with any exception. That would allow us to just make this a null check.